### DISCUSSION TITLE
Problems with Azure authentication provider

### DISCUSSION THREAD

**Discussion by fasolato-omicron** [requarks/wiki#6200] [Error / Bug Report] [answered]
I am having problems using AzureAD as a provider for our authentication. The error I receive is the same "wrong username or password" that I have already read in other threads in this discussion forum.

I am opening a new thread because all the others were missing an information (I think).

First of all I tried all the other solutions and no one worked, and I tested the Azure configuration in another application and it is working correctly.

To solve the issue I tried starting the wiki locally on my machine and debug the process (I cloned this repo and am using the master branch).
Reading another thread I started with the file `server/modules/authentication/azure/authentication.js`, line 40. This is the function that should be called when Azure returns to the callback url passed by the wiki. The problem here is that this code is never executed: the debugger stops right after that function, in the file `server/models/users.js`, line 316, where the parameter `user` is false, as it's never been initialized in the first place.

I also tried with the local auth strategy, and the corresponding function in the file `server/modules/authentication/local/authentication.js` is correctly called and the local user is correctly recognized.

I do not know if the function is not called because of a configuration issue or something else, but I suspect that all the issues with Azure and its authentication are related to this problem. Either it's a wiki problem, or a configuration issue. If the latter, I think the documentation regarding Azure authentication could be updated because it's a bit scarce and in some points downright wrong (the callback url schema, for example).

Apart from that, keep up the good work, the wiki is a nice product.

**Comment by fasolato-omicron** [comment, marked as answer]
We solved our problem, appharently the documentation is not 100% clear.

I could not reach in debug the code I expected to reach, because the Application on AzureAD side was not configured correctly. The application must be configured to allow implicit and hybrid flows, otherwise Azure replies with an error message and status, so the wiki does not try to extract the principal information from the response, so my breakpoint is never reached.

The correct configuration (under Authentication in Azure AD) is like this:

![image](https://user-images.githubusercontent.com/95283408/221887394-8de4fafc-730f-445a-9478-4ec247c646c2.png)

**Reply by SimonCoutureCSTJ** [reply]
@fasolato-omicron Did not work for me and still have the login issue. Did yout do anything else beside what you mentionned?

**Reply by mittkkoo** [reply]
Well, my Azure authentication works with only "ID tokens (used for implicit and hybrid flows)" checked... I'm not experiencing any problems.
